### DISCUSSION TITLE
TimedSubdomainModifier at time zero

### DISCUSSION THREAD

**Discussion by Wendy-Ji** [idaholab/moose#28125] [Q&A General]
### Check these boxes if you have followed the posting rules.

- [X] Q&A General is the most appropriate section for my question
- [X] I have consulted the posting Guidelines on the Discussions front page
- [X] I have searched the Discussions forum and my question has not been asked before
- [X] I have searched the MOOSE website and the documentation does not answer my question
- [X] I have formatted my post following the posting guidelines (screenshots as a last resort, triple back quotes around pasted text)

### Question

Hello, 

I would like to check whether this was intended. If the time is set to zero, then the subdomain change does not occur. Below is an example file, where subdomains 2 and 3 are changing to subdomain 1 at times 0 and 1 respectively. Subdomain 3 changes, but subdomain 2 remains the same for the whole simulation. Was this because a subdomain change for a whole block shouldn't occur at time zero? 

Thank you.
```
[Problem]
  solve = false
[]

[Mesh]
  [gen]
    type = GeneratedMeshGenerator
    dim = 2
    xmax = 3
    ymax = 1
    nx = 3
    ny = 1
  []
  [left]
    type = SubdomainBoundingBoxGenerator
    input = 'gen'
    block_id = 1
    bottom_left = '0 0 0'
    top_right = '1 1 0'
  []
  [middle]
    type = SubdomainBoundingBoxGenerator
    input = 'left'
    block_id = 2
    bottom_left = '1 0 0'
    top_right = '2 1 0'
  []
  [right]
    type = SubdomainBoundingBoxGenerator
    input = 'middle'
    block_id = 3
    bottom_left = '2 0 0'
    top_right = '3 1 0'
  []
[]

[Variables]
  [dummy]
  []
[]

[UserObjects]
  [block_change]
    type = TimedSubdomainModifier
    times = '0 1'
    blocks_from = '2 3'
    blocks_to = '1 1'
    execute_on = 'INITIAL TIMESTEP_BEGIN'
  []
[]

[Executioner]
  type = Transient
  dt = 1
  end_time = 2
[]

[Outputs]
  exodus = true
[]
```

**Comment by GiudGiud** [comment]
Hello

I think this might have been a miss. I ll check and create an issue

If you need this, you can set a start_time to the simulation before 0

Guillaume

**Reply by jmeier** [reply]
Is there any news on this yet?

**Reply by GiudGiud** [reply]
I created an issue
https://github.com/idaholab/moose/issues/28132

the workaround being so simple, it's not a priority to change this
